pkg/controller/job/job_controller.go

```diff
@@ -627,8 +629,10 @@ func (jm *Controller) syncJob(key string) (forget bool, rErr error) {
 					// consistent with resuming a Job created in the suspended state.
 					// (ActiveDeadlineSeconds is interpreted as the number of seconds a
 					// Job is continuously active.)
-					now := metav1.Now()
-					job.Status.StartTime = &now
+					if !setStartTime { // prevent StartTime to be re-set, then ActiveDeadlineSeconds lose efficacy
```

Comment:
Don't change this bit, this was described in https://github.com/kubernetes/enhancements/tree/master/keps/sig-apps/2232-suspend-jobs#design-details and this is working as it should.

Reply:
This still applies.